lib/artifact-search.js

```diff
@@ -142,6 +145,21 @@ module.exports = (bluebird, error, logger, requestOptions, requestPromise, respo
                     // Pass the reason response to the error handler.
                     responseHandler.handleErrorResponse(reason);
                 });
+            }, (rejectArr) => {
+                var errorMessage;
+
+                errorMessage = [];
+                rejectArr.forEach((err) => {
+                    if (!errorMessage.includes(err.message)) {
+                        errorMessage.push(`${err.code}\n`);
+                    }
+
+                    if (err instanceof Error) {
+                        errorMessage.push(`${err.message}\n`);
+                    }
+                });
+
+                return errorMessage.join();
```

Comment:
This doesn't keep the error rejected. This will actually resolve the promise. Also it is not an error. Also no errors should be duplicate. I see that they will be, but that doesn't help the user that much. Instead the error should contain the query that errored. Also We shouldn't need to check if its an instance of Error. It should always be a `ShelfError` but even if it was not we should be consistent with the values returned.

After that is done you should just be able to do map.

```
messageList = rejectionList.map((err) => {
    return err.message;
});
message = messageList.join("\n");
```